zenoh/src/net/routing/dispatcher/resource.rs

```diff
@@ -178,6 +178,15 @@ impl SessionContext {
     }
 }
 
+pub(crate) fn merge_qabl_infos(
+    mut this: QueryableInfoType,
+    info: &QueryableInfoType,
+) -> QueryableInfoType {
+    this.complete = this.complete || info.complete;
+    this.distance = std::cmp::min(this.distance, info.distance);
```

Comment:
As you suggested in another discussion it would be better to keep the smallest distance of the completes Queryables if there are some.

```suggestion
    this.distance = match (this.complete, info.complete) {
        (true, true) | (false, false) => std::cmp::min(this.distance, info.distance),
        (true, false) => this.distance,
        (false, true) => info.distance,
    };
```

Reply:
done